addons/board/static/src/js/dashboard.js

```diff
@@ -347,6 +347,10 @@ core.form_tag_registry
 FavoriteMenu.include({
     start: function () {
         var self = this;
+        if(self.action_id === undefined) {
+            this.$('.o_add_to_dashboard_link').remove();
```

Comment:
This line is not necessary because the 'Add to dashboard' link is appended to the $el later (at line 360).